Source/EmbeddedBoundary/ParticleScraper.H

```diff
@@ -172,10 +172,14 @@ scrapeParticles (PC& pc, const amrex::Vector<const amrex::MultiFab*>& distance_t
 
                 // the closest point on the surface to pos is pos - grad phi(pos) * phi(pos)
                 amrex::RealVect pos;
+#if (defined WARPX_DIM_3D)
                 pos[0] = xp - normal[0]*phi_value;
                 pos[1] = yp - normal[1]*phi_value;
                 pos[2] = zp - normal[2]*phi_value;
-
+#elif (defined WARPX_DIM_XZ)
+                pos[0] = xp - normal[0]*phi_value;
+                pos[1] = yp - normal[1]*phi_value;
+#endif
```

Comment:
@dpgrote do you see on first view what the corresponding RZ logic would be? :)
Let's fix all three (four #2307) geometries at once.

Reply:
In RZ, this would need to use the raw data (from the AOS) since this would need the `r` and `z` values, since the boundaries would presumably be defined in `r,z`.

Reply:
@ax3l I fixed the `yp/zp` mixup.

@dpgrote I'll be happy to add the `RZ` case if I know what it needs to say, or directly accept a 'suggestion' here on the PR

Reply:
Does the code even support EB in RZ configurations right now?

Reply:
Good question. If not, let's add an assert here that the person that adds it will find the to-be-updated location via an assert instead of a segfault :D

Reply:
Aborts in the RZ case for now. Soimeone can address that later if it comes up.